src/parser.js

```diff
@@ -265,7 +272,9 @@ var parsers = {
   },
 
   msdos: function(entry) {
-    var group = entry.match(RE_DOSEntry);
+    var isDir = (entry.indexOf('<DIR>') > 0);
+    var regx =  isDir ? RE_DOSEntry : RE_DOSEntry2;
```

Comment:
No need to create `regx` variable. Please put `isDir ? RE_DOSEntry : RE_DOSEntry2` inside `entry.match` below.